File: PomodoraBack/DTOs/UserDto.cs
namespace PomodoraBack.DTOs
{
    public class UserDto
    {
        public string UserId { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Surname { get; set; } = string.Empty;
        public string Nickname { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public bool CurrentStatus { get; set; }
        public decimal TotalPoints { get; set; }
        public DateTime LastSeen { get; set; }
        public DateTime? DeletedAt { get; set; }
        
        /// <summary>
        /// Kullanıcının gerçek zamanlı online durumu.
        /// CurrentStatus = true VE son 5 dakika içinde aktivite varsa true döner.
        /// </summary>
        public bool IsOnline 
        { 
            get 
            {
                var fiveMinutesAgo = DateTime.UtcNow.AddMinutes(-5);
                return CurrentStatus && LastSeen > fiveMinutesAgo;
            }
        }
    }
}

File: frontend/lib/features/profile/models/user_dto.dart
class UserDto {
    final String userId;
    final String name;
    final String surname;
    final String nickname;
    final String email;
    final bool currentStatus;
    final double totalPoints;
    final DateTime lastSeen;
    final bool isOnline;

    const UserDto({
        required this.userId,
        required this.name,
        required this.surname,
        required this.nickname,
        required this.email,
        required this.currentStatus,
        required this.totalPoints,
        required this.lastSeen,
        required this.isOnline,
    });

    factory UserDto.fromJson(Map<String, dynamic> json) {
        return UserDto(
            userId: json['userId'] ?? '',
            name: json['name'] ?? '',
            surname: json['surname'] ?? '',
            nickname: json['nickname'] ?? '',
            email: json['email'] ?? '',
            currentStatus: json['currentStatus'] ?? false,
            totalPoints: (json['totalPoints'] ?? 0).toDouble(),
            lastSeen: DateTime.parse(json['lastSeen']),
            isOnline: json['isOnline'] ?? false,
        );
    }

    String get displayName {
        final full = ('$name $surname').trim();
        return full.isNotEmpty ? full : nickname;
    }
}

